be/src/storage/delta_writer.cpp

```diff
@@ -33,8 +33,8 @@
 
 namespace starrocks {
 
-OLAPStatus DeltaWriter::open(WriteRequest* req, MemTracker* mem_tracker, DeltaWriter** writer) {
-    *writer = new DeltaWriter(req, mem_tracker, StorageEngine::instance());
+OLAPStatus DeltaWriter::open(WriteRequest* req, MemTracker* mem_tracker, std::shared_ptr<DeltaWriter>* writer) {
+    *writer = std::shared_ptr<DeltaWriter>(new DeltaWriter(req, mem_tracker, StorageEngine::instance()));
```

Comment:
std::make_shared

Reply:
the `DeltaWriter`'s constructor is private, also copy assignment is deleted, so i can only use this form. this is exception safe, because if exception throws, shared_ptr will call `delete T`.

Reply:
https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr